tmk_core/protocol/report.h

```diff
@@ -327,6 +327,8 @@ static inline uint16_t KEYCODE2CONSUMER(uint8_t key) {
             return AC_DESKTOP_SHOW_ALL_WINDOWS;
         case KC_LAUNCHPAD:
             return AC_SOFT_KEY_LEFT;
+        case KC_GLOBE:
+            return AC_NEXT_KEYBOARD_LAYOUT_SELECT;
```

Comment:
Looks like this KC_GLOBE does exactly that, and only suitable for switching layouts, not really sending Globe codes.

Reply:
Strictly speaking, you are right, but in practice, on a Mac, it operates as a key.
- pressed alone, it triggers the emoji picker (or what is configured in the Mac settings for the Keyboard)
- it can be used as a "modifier" with specific keys, for instance 
  - Globe+A to toggle the dock when hidden, 
  - Globe+C to toggle control center, 
  - Globe+F to toggle full screen, 
  - Globe+H to show the desktop, 
  - Globe+N to toggle Notification Center, 
  - Globe+Q launch notes.
Unfortunately, there are other cases where I have yet to have success. In particular, Globe+Arrow keys combinations, but I typically use 60% keyboards, so arrow keys are hard to test with those.

Reply:
If the keycode works on Windows/Linux too (but as layout select), then it would be worth having appropriate aliases too.

Reply:
I don't have a pure Windows or Linux machine, but I can investigate using a virtual machine. I'll see whether I have time tomorrow to try that.

Reply:
Windows does not seem to react to the `AC Next Keyboard Layout Select` usage, but I can see it coming in through Wireshark.

Reply:
Yes, I tried this morning while at work on Windows 10, and nada, no reaction to the key. I'll try Linux later; I suspect I'll see the same thing.

Reply:
Sorry for the delay. I have tried the globe key under a virtual Linux running on my Mac. But naively, I did not realize that the key would be handled by the Mac, and thus, I did not get any different behavior than for regular Mac apps. I would need someone with a Linux box to try natively. Sorry about that.

Reply:
If it helps, the kernel input handling code does know about this consumer usage, but I suppose it's up to the higher layers as to what happens with it:
https://elixir.bootlin.com/linux/v6.12.1/source/drivers/hid/hid-input.c#L1234

And you should be able to passthrough the USB device directly to the VM (at least on Virtualbox this is possible) to stop the host OS capturing it.

Reply:
You're right about the VM. Indeed, this is what happens when I connect to work via Citrix (the reason so many of my colleagues have keyboard problems). I'll try it again by configuring Parallels properly. I'll do a search, but by any chance, do you know any Linux software that would capture keyboard events like Karabiner on the Mac?

Reply:
I succeeded in passing all my keypresses to the Linux session running in VM. I've monitored the keypresses using `xev`, which is pretty basic but does the job. Like with Windows, the result is no report of any key being pressed when using either the Globe on its own or in combination with another key.

Reply:
I figured that might be the case, considering X keysyms only go up to 247 (for a total of 248 possible values); `KEY_KBD_LAYOUT_NEXT` has a value of 584. `evdev` might have better info, but even so, I'm not sure what the mechanism(s) would be, if any, to get the key to do something useful.

Reply:
Actually [xkeyboard-config](https://gitlab.freedesktop.org/xkeyboard-config/xkeyboard-config) has [support for `KEY_KBD_LAYOUT_NEXT` in `symbols/inet`](https://gitlab.freedesktop.org/xkeyboard-config/xkeyboard-config/-/commit/2a0c538c518becc49aedd66dc21460418af1dcd8#f3d4a1d386e363c7fedc3c05285f0343c340e9cf_233_356) since `xkeyboard-config-2.33` (the `<I592>` keycode (`linux_input_keycode + 8`, as usual) gets mapped to `ISO_Next_Group`).  The problem is that the X protocol does not support keycode values above 255, therefore that mapping would work only in Wayland.

Reply:
Well, clearly, there is atm only a use case for macOS but it is a big one for Apple users. I'd wish to know how to make all the other combinations work but that's already a beginning that would benefit this group of users.